### ISSUE TITLE
Lightning gun lockup

### ISSUE THREAD

**Issue by jdolan** [jdolan/quetoo#111]
```
Program received signal SIGINT, Interrupt.
0x00007fffecb04387 in G_Find (from=0x1585190, from@entry=0x0,
    field=field@entry=0,
    match=match@entry=0x7fffecb0de6a "G_LightningProjectile") at g_util.c:110
110             for (; from < &g_game.entities[ge.num_entities]; from++) {
(gdb) backtrace
#0  0x00007fffecb04387 in G_Find (from=0x1585190, from@entry=0x0,
    field=field@entry=0,
    match=match@entry=0x7fffecb0de6a "G_LightningProjectile") at g_util.c:110
#1  0x00007fffecb0684d in G_FireLightning (ent=0x15835b8) at g_weapon.c:522
#2  0x00007fffecaf5b0d in G_ClientBeginFrame (ent=ent@entry=0x15835b8)
    at g_client.c:1378
#3  0x00007fffecb014d3 in G_Frame () at g_main.c:806
#4  0x0000000000405a03 in Sv_RunGameFrame () at sv_main.c:621
#5  Sv_Frame (msec=msec@entry=1) at sv_main.c:753
#6  0x0000000000403827 in Frame (msec=1) at main.c:250
#7  main (argc=9, argv=0x7fffffffebd8) at main.c:312
```

**Comment by jar3817** [comment]
Maybe related to the fact that two players (you and that other guy) were both using LG on each other...maybe not.

**Comment by jdolan** [comment]
Resolved.

**Comment by jar3817** [comment]
Still broken. Two players crossing streams and server locked up. Load average spiked to 100% and stayed there until I killed the quetoo-dedicated process.

âLinh connected                                                                
âba1b0a connected                                                              
âba1b0a connected                                                              
âLinh wants some                                                               
âba1b0a wants some                                                             
âLinh was tased by ba1b0a's lightning                                          
âba1b0a: toggle_console                                                        
â                                                                              
Program received signal SIGINT, Interrupt.
0x00007ffff17d6a77 in G_Find (from=0x149bae0, from@entry=0x0,
    field=field@entry=0,
    match=match@entry=0x7ffff17e0a9a "G_LightningProjectile") at g_util.c:121
121     g_util.c: No such file or directory.
Missing separate debuginfos, use: debuginfo-install SDL2-2.0.3-5.fc21.x86_64 SDL2_image-2.0.0-7.fc21.x86_64 SDL2_mixer-2.0.0-7.fc21.x86_64 bzip2-libs-1.0.6-14.fc21.x86_64 elfutils-libelf-0.163-4.fc21.x86_64 elfutils-libs-0.163-4.fc21.x86_64 glib2-2.42.2-1.fc21.x86_64 glibc-2.20-8.fc21.x86_64 jbigkit-libs-2.1-2.fc21.x86_64 libgcc-4.9.2-6.fc21.x86_64 libjpeg-turbo-1.3.1-5.fc21.x86_64 libpng-1.6.10-3.fc21.x86_64 libstdc++-4.9.2-6.fc21.x86_64 libtiff-4.0.3-20.fc21.x86_64 libwebp-0.4.4-1.fc21.x86_64 lzma-sdk457-4.57-8.fc21.x86_64 ncurses-libs-5.9-16.20140323.fc21.x86_64 nss-mdns-0.10-15.fc21.x86_64 physfs-2.0.3-4.fc21.x86_64 systemd-libs-216-25.fc21.x86_64 xz-libs-5.1.2-14alpha.fc21.x86_64 zlib-1.2.8-7.fc21.x86_64
(gdb) backtrace
#0  0x00007ffff17d6a77 in G_Find (from=0x149bae0, from@entry=0x0,
    field=field@entry=0,
    match=match@entry=0x7ffff17e0a9a "G_LightningProjectile") at g_util.c:121
#1  0x00007ffff17d931d in G_FireLightning (ent=0x1485940) at g_weapon.c:683
#2  0x00007ffff17c81cd in G_ClientBeginFrame (ent=ent@entry=0x1485940)
    at g_client.c:1396
#3  0x00007ffff17d3b83 in G_Frame () at g_main.c:806
#4  0x0000000000405ce0 in Sv_RunGameFrame () at sv_main.c:623
#5  Sv_Frame (msec=msec@entry=3) at sv_main.c:752
#6  0x0000000000403ac9 in Frame (msec=3) at main.c:230
#7  main (argc=<optimized out>, argv=0x7fffffffe5f8) at main.c:286
(gdb)

[root@qt ~]# uptime
 12:04:15 up 224 days,  5:47,  1 user,  load average: 0.32, 0.80, 0.97

**Comment by jdolan** [comment]
I think this must have been an older build. Both instances of the unbounded loop to find the current lighting projectile have been fixed. I just started a listen server with cheats, and connected a 2nd instance of the game locally. I gave both clients 9999 bolts, and typed +attack in one console, while running around in the other instance of the client intermittently firing the lightning at will. Tested in this way for 2-3 minutes straight. I really don't think this bug exists anymore.
